hosted-cluster-config-operator/operator/config.go

```diff
@@ -198,6 +204,32 @@ func (c *HostedClusterConfigOperatorConfig) TargetKubeInformersForNamespace(name
 	return informer
 }
 
+func (c *HostedClusterConfigOperatorConfig) ControlPlaneCluster() cluster.Cluster {
+	if c.cpCluster == nil {
```

Comment:
Could we try to avoid this awful contructor pattern and simply use a public `CPCluster` property that we initialize in `NewHostedClusterConfigOperatorConfig` ?

Reply:
Same comment for other things like `HCPName`, I fail to see any value in making these private and then adding a getter and it adds mental overhead when trying to understand the code.

Reply:
agree... it feels really burdensome to add a new field. However, can we do it when we finish consolidating controllers into the resources controller so we can avoid unnecessary churn?

Reply:
I guess what I am saying is, I don't expect you to change the pattern for existing code, but IMHO we should not repeat that for new code and do it properly, to avoid adding code that we then change again shortly after